### PR TITLE
Bug 1320699 - Add experiments ETL job

### PR DESCRIPTION
**Pull request by fbertsch**
@mreid-moz r?

### CODE REVIEW

File: etl/experiments.kp/knowledge.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+# on 'yesterday' by default.
+yesterday = dt.strftime(dt.utcnow() - timedelta(1), "%Y%m%d")
+target_date = environ.get('date', yesterday)
+```
+
+```python
+from moztelemetry.dataset import Dataset
+
+sample_rate = environ.get('sample', 1)
+pings = Dataset.from_source("telemetry") \
```
Comment (mreid-moz) [inline etl/experiments.kp/knowledge.md]: You can use the "telemetry-experiments" dataset which has the following dimensions:
- submissionDate
- docType
- activeExperimentId

You'll need to filter on `appName == Firefox` later though.

File: etl/experiments.kp/knowledge.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        channel_agg['errors'].append('{}: {}'.format(e.__class__, str(e)))
+    
+    return channel_agg
+
+def channel_channel_agg(channel_agg_1, channel_agg_2):
+    """Aggregate a channel with a channel"""
+    channel_agg_1['total'] += channel_agg_2['total']
+    channel_agg_1['errors'] += channel_agg_2['errors']
+    
+    for id, exp in channel_agg_2['experiments'].iteritems():
```
Comment (mreid-moz) [inline etl/experiments.kp/knowledge.md]: ditto, please change throughout.

File: etl/experiments.kp/knowledge.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+from copy import deepcopy
+
+### Aggregation functions, Spark job, output file creation ###
+
+def channel_ping_agg(channel_agg, ping):
+    """Aggregate a channel with a ping"""
+    channel_agg['total'] += 1
+
+    try:
+        for item in (ping.get("log") or []):
```
Comment (fbertsch) [inline etl/experiments.kp/knowledge.md]: This isn't the same, right? The ping will always have "log", just sometimes it's an empty dict (resulting from get_pings_properties). Any case where "log" is None, empty, missing, etc. I want it to be [].

Comment (mreid-moz) [inline etl/experiments.kp/knowledge.md]: Gotcha!

File: etl/experiments.kp/knowledge.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+def get_empty_channel():
+    return deepcopy(channel)
+```
+
+```python
+import gzip
+import ujson
+import requests
+
+COUNTS_JSON_URI = "https://pipeline-cep.prod.mozaws.net/dashboard_output/analysis.frank.main_channel_counts..txt"
```
Comment (mreid-moz) [inline etl/experiments.kp/knowledge.md]: Please add some comments to describe this URL and the associated code :)

**Comment by fbertsch** [comment]
@mreid-moz added comments, RFAL

**Pull request merged**